pennylane/math/utils.py

```diff
@@ -281,6 +281,9 @@ def requires_grad(tensor, interface=None):
     if interface == "jax":
         import jax
 
-        return isinstance(tensor, jax.interpreters.ad.JVPTracer)
+        # TODO: modify this once we have a way of distinguishing jax arrays to
+        # be differentiable or not
+        # return isinstance(tensor, jax.interpreters.ad.JVPTracer)
+        return isinstance(tensor, jax.numpy.ndarray)

```

Comment:
Can you remind me why we had to change this?

Reply:
Oh, is this because checking for traced arrays only works in non-JIT mode?

Reply:
I got into cases where JVPTracer objects were mixed with simple DeviceArray objects in a `JacobianTape`. This lead to an error here:
https://github.com/PennyLaneAI/pennylane/blob/6d93cc580d4e8860a0b3fc8cc42e81198399e665/pennylane/tape/tape.py#L861

Removing the check produced incorrect results.

Reply:
This has now been reverted and addressed the specific test case here:
https://github.com/PennyLaneAI/pennylane/pull/1634#discussion_r714074895